boards/silabs/radio_boards/slwrb4180b/slwrb4180b.dts

```diff
@@ -0,0 +1,146 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright (c) 2024 Capgemini
+ * Author: Arunmani Alagarsamy <arunmani.a@capgemini.com>
+ *
+ */
+
+/dts-v1/;
+#include <silabs/efr32mg21a020f1024im32.dtsi>
+#include <zephyr/dt-bindings/input/input-event-codes.h>
+#include "slwrb4180b-pinctrl.dtsi"
+
+/ {
+	model = "Silicon Labs BRD4180B (Wireless Gecko 21 Radio Board)";
+	compatible = "silabs,slwrb4180b", "silabs,efr32mg21";
+
+	chosen {
+		zephyr,console = &usart0;
+		zephyr,shell-uart = &usart0;
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+		zephyr,code-partition = &slot0_partition;
+	};
+
+	/* These aliases are provided for compatibility with samples */
+	aliases {
+		led0 = &led0;
+		led1 = &led1;
+		sw0 = &button0;
+		sw1 = &button1;
+		watchdog0 = &wdog0;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		led0: led_0 {
+			gpios = <&gpiod 2 0>;
+			label = "LED 0";
+		};
+		led1: led_1 {
+			gpios = <&gpiod 3 0>;
+			label = "LED 1";
+		};
+	};
+
+	buttons {
+		compatible = "gpio-keys";
+		button0: button_0 {
+			gpios = <&gpiob 0 GPIO_ACTIVE_LOW>;
+			label = "User Push Button 0";
+			zephyr,code = <INPUT_KEY_0>;
+		};
+		button1: button_1 {
+			gpios = <&gpiob 1 GPIO_ACTIVE_LOW>;
+			label = "User Push Button 1";
+			zephyr,code = <INPUT_KEY_1>;
+		};
+	};
+};
+
+&cpu0 {
+	clock-frequency = <38400000>;
+};
+
+&usart0 {
+	current-speed = <115200>;
+	pinctrl-0 = <&usart0_default>;
+	pinctrl-names = "default";
+	status = "okay";
+};
+
+&rtcc0 {
+	prescaler = <1>;
+	status = "okay";
```

Comment:
@jhedberg, I am not sure the Zephyr policy here. I assume all the non-essential devices should be declared with `status = "disabled"`. The application is in charge of enabling them.